FR #1055: Adding language to provide expected behavior of deprecated columns

Matt-Cowsert opened the pull request:
This PR focuses on clarifying the expected behavior of data generators for deprecated capabilities, especially once the FOCUS project removes a capability from the FOCUS specification. 

Additionally, this PR adds an expanded definition for removal to provide additional details top data generators and practitioners on how to track changes over time.

Matt-Cowsert committed:
Adding language to provide expected behavior of deprecated columns in change management guidelines

shawnalpay commented on guidelines/spec-change-guidelines.md:
```diff
@@ -51,13 +51,16 @@ The FOCUS specification requires that removal actions are preceded with a deprec
 * Incompatible if removal results in the loss of support for a supported feature.
 
+#### Expected Behavior for Deprecated Capabilities
+Providers are expected to remove deprecated capabilities at the same time that the FOCUS specification does. If providers choose to retain a capability (e.g., column) after its been removed from the FOCUS specification, they must explain the rationale in their FOCUS documentation (e.g., conformance gap report). 
```
This verbiage also needs to be augmented with clarification between "deprecated" and "dropped", per AI #1504.

```suggestion
Data generators are expected to remove dropped capabilities at the same time that the FOCUS specification does. If a data generator chooses to retain a capability (e.g., column) after it has been removed from the FOCUS specification, they must explain the rationale in their FOCUS documentation (e.g., conformance gap report). 
```

shawnalpay reviewed (changes requested):
Please:

* Address the suggestion I left.
* Add a summary of the scope of the change in the parent post.

Matt-Cowsert committed:
Update guidelines/spec-change-guidelines.md

Co-authored-by: Shawn Alpay <77511110+shawnalpay@users.noreply.github.com>

Matt-Cowsert committed:
Update spec-change-guidelines.md

Added definition for the dropped status

Matt-Cowsert committed:
Update spec-change-guidelines.md

expanded the definition of removal and accounted for the expected behavior of data generators post-deprecation

Matt-Cowsert commented:
Updated scope of this pull request after conferring with @shawnalpay on 2025-10-07

shawnalpay commented on guidelines/spec-change-guidelines.md:
```diff
@@ -51,13 +51,16 @@ The FOCUS specification requires that removal actions are preceded with a deprec
 #### Deprecation Retention Period
 Once a capability or column is designated as deprecated in a published version of the FOCUS specification:
-* It MUST remain in the specification for a minimum of 3 releases (inclusive of the version where the deprecation is announced) before it is eligible for removal.
+* It MUST remain in the specification for a minimum of 2 releases (inclusive of the version where the deprecation is announced; e.g., announced in 1.3 and removed in 1.4) before it is eligible for removal.
```
This line feels strange to me.  

Let's remove the parentheses for a moment:

> It MUST remain in the specification for a minimum of 2 releases before it is eligible for removal.

Then, in the example, the deprecation is carried for one release (1.3), then removed in the next release (1.4).

This either needs to change to be:

> It MUST remain in the specification for a minimum of 1 release before it is eligible for removal.

Or some other option that talks about "minimum of 2 releases" being _inclusive_ of the release that removes it.

Matt-Cowsert replied:
I've re-written the requirement to reduce ambiguity, let me know what you think.

shawnalpay replied:
Looks good!  Resolving.

shawnalpay committed:
Consistency pass

shawnalpay commented on guidelines/spec-change-guidelines.md:
```diff
@@ -28,36 +28,43 @@ A changelog is provided specifying helpful information to consumers of the FOCUS
+* It MUST remain in the specification for a minimum of 2 releases (inclusive of the version where the deprecation is announced; e.g., announced in 1.3 and removed in 1.4) before it is eligible for removal.
 * Exceptions (e.g., legal, security, or privacy requirements) MAY justify earlier removal; such exceptions MUST be explicitly documented in the changelog with rationale.
 
 Removal of a deprecated feature or column is classified as:
```
Is this the right language?  Are we classifying it only once it's been removed?  Or are we classifying it once it's been deprecated, prior to removal?

Matt-Cowsert replied:
I've re-written the requirement to reduce ambiguity. Let me know what you think. The expectation is that we're making this classification at the point of deprecation, not at the point of removal.

shawnalpay replied:
Looks good!  Resolving.

shawnalpay commented on guidelines/spec-change-guidelines.md:
```diff
@@ -28,36 +28,43 @@ A changelog is provided specifying helpful information to consumers of the FOCUS
 * Migration Compatible if the same supported feature is fully supported by another feature or column in the specification.
 * Incompatible if removal results in the loss of support for a supported feature.
 
+#### Expected Behavior for Deprecated Capabilities
```
```suggestion
#### Expected Behavior for Removed Capabilities
```

Matt-Cowsert committed:
Update spec-change-guidelines.md

updated to address Shawn's concerns on confusing language

Matt-Cowsert committed:
Apply suggestion from @shawnalpay

Co-authored-by: Shawn Alpay <77511110+shawnalpay@users.noreply.github.com>

Matt-Cowsert committed:
Update spec-change-guidelines.md

Added clarification on when a capability inherits its change type classification

shawnalpay reviewed (approved):
Approved!  Nice job @Matt-Cowsert.

shawnalpay committed:
Added missing period

shawnalpay commented on guidelines/spec-change-guidelines.md:
```diff
@@ -68,11 +76,11 @@ The addition of a new feature or column to the FOCUS specification.
 ### Removal
 
-The removal of a feature or column from the FOCUS specification.
+A feature or column removed from the specification after completing deprecation requirements or with documented exception rationale. The changelog specifies the removal version, deprecation timeline, and compatibility classification.
```
@Matt-Cowsert @rileyjenk 
I expect this verbiage to be applied and reviewable, either in #1139 or #1494, before we can approve either this PR or #1139.  I am not comfortable with waiting for consistency review to handle this; we are setting an important precedent, and we need time to discuss it ahead of approval.

shawnalpay replied:
@rileyjenk to update #1494 to include deprecation information in changelog.

shawnalpay replied:
This was done!  Resolving.

shawnalpay committed:
Merge branch 'working_draft' into 1055-change-management-improvements

Matt-Cowsert commented:
we closed all open conversations and this PR is ready for member approval tomorrow.

jpradocueva commented:
PR was approved during the Members’ Meeting on October 16.
